cou/steps/plan.py

```diff
@@ -349,15 +349,62 @@ def generate_common_plan(
                 raise_on_blocked=True,
             ),
         )
-    )
+    ]
     if args.backup:
-        plan.add_step(
+        steps.append(
             PreUpgradeStep(
                 description="Backup mysql databases",
                 coro=backup(analysis_result.model),
             )
         )
-    return plan
+
+    return steps
+
+
+def _get_post_upgrade_steps(analysis_result: Analysis, args: CLIargs) -> list[PostUpgradeStep]:
+    """Get the post-upgrade steps.
+
+    :param analysis_result: Analysis result
+    :type analysis_result: Analysis
+    :param args: CLI arguments
+    :type args: CLIargs
+    :return: List of post-upgrade steps.
+    :rtype: list[PreUpgradeStep]
+    """
+    steps = []
+    if args.upgrade_group in {DATA_PLANE, None}:
+        print("here")
+        steps.extend(_get_ceph_mon_post_upgrade_steps(analysis_result.apps_data_plane))
+
+    return steps
+
+
+def _get_ceph_mon_post_upgrade_steps(apps: list[OpenStackApplication]) -> list[PostUpgradeStep]:
+    """Get the post-upgrade step for ceph-mon, where we check the require-osd-release option.
+
+    :param apps: List of OpenStackApplication.
+    :type apps: list[OpenStackApplication]
+    :return: The post-upgrade step.
+    :rtype: list[PreUpgradeStep]
+    :raises NotSupported: When two ceph-mon are found at the model.
+    """
+    ceph_mons_apps = [app for app in apps if isinstance(app, CephMon)]
+    if num_ceph_mon_apps := len(ceph_mons_apps) > 1:
+        raise NotSupported(
+            f"Deployment with {num_ceph_mon_apps} ceph-mon applications is not supported."
+        )
+
+    steps = []
+    for app in ceph_mons_apps:
```

Comment:
Making steps a list and also iterating through `ceph_mons_apps` doesn't look like a good approach because it gives user a false impression that more than one ceph-mon apps in deployment can be handled and will generate multiple steps, where in fact steps can only be empty or with 1 step added.

Reply:
Actually we will handle multiple ceph-mon, I just forgot to remove that raise.